Source/Queue/Queue.cs

```diff
@@ -49,6 +49,53 @@ public static void TryDequeue( ResearchNode node )
         {
             if ( _instance._queue.Contains( node ) )
                 Dequeue( node );
+
+        public static bool TryToMove(ResearchNode node)
+        {
+            var dropPosition = Event.current.mousePosition;
+            // Check if Mouse is Outside node
+            if (node.RealQueueRect.Contains(dropPosition)) return false;
+
+            // Order Queue by distance to mouse 
+            var queueByDistance = _instance._queue.OrderBy(item => Vector2.Distance(item.RealQueueRect.center, dropPosition));
+            var nearestItem = queueByDistance.First();
+            var indexToPlace = _instance._queue.IndexOf(nearestItem);
+
+            // Move node to new Position
+            _instance._queue.Remove(node);
+            _instance._queue.Insert(indexToPlace, node);
```

Comment:
looks like this will fail if node's `index < indexToPlace`, you'll effectively insert at indexToPlace +1

Reply:
I tested it, and didn't get any error in any case.

IndexToPlace is saved before node is removed from list and indexes move by -1.
So if you have 10elements and want to place at last position then insert will act as Add.

example from C# Interactive
```
Loading context from 'CSharpInteractive.rsp'.
Type "#help" for more information.
> var q = new List<int>() {0,1,2,3,4,5,6,7,8,9};
> q.Count
10
> var indexToPlace = q.IndexOf(9);
> var node = q[5];
> q.Remove(node);
> q.Insert(indexToPlace,node);
> q.Count
10
> q[indexToPlace]
5
> foreach(var n in q){Console.WriteLine(n);}
0
1
2
3
4
6
7
8
9
5
```

Reply:
Ugh i replied inside code:

I tested it, and didn't get any error in any case.

IndexToPlace is saved before node is removed from list and indexes move by -1.
So if you have 10elements and want to place at last position then insert will act as Add.

example from C# Interactive

Loading context from 'CSharpInteractive.rsp'.
Type "#help" for more information.
> var q = new List<int>() {0,1,2,3,4,5,6,7,8,9};
> q.Count
10
> var indexToPlace = q.IndexOf(9);
> var node = q[5];
> q.Remove(node);
> q.Insert(indexToPlace,node);
> q.Count
10
> q[indexToPlace]
5
> foreach(var n in q){Console.WriteLine(n);}
0
1
2
3
4
6
7
8
9
5

Reply:
wow, that was fast.

If you're online now, wanna chat on discord? I'm Fluffy on the unofficial official RimWorld discord: https://discord.gg/rimworld